core/block/source/source.go

```diff
@@ -282,28 +316,57 @@ func (s *source) PushChange(params PushChangeParams) (id string, err error) {
 			Errorf("change size (%d bytes) is above the limit of %d bytes", len(data), changeSizeLimit)
 		return "", err
 	}
-	addResult, err := s.ObjectTree.AddContent(context.Background(), objecttree.SignableChangeContent{
+	content := objecttree.SignableChangeContent{
 		Data:        data,
 		Key:         s.accountService.Account().SignKey,
-		IsSnapshot:  c.Snapshot != nil,
+		IsSnapshot:  change.Snapshot != nil,
 		IsEncrypted: true,
-	})
+	}
+
+	if dataType != "" {
```

Comment:
is this condition necessary? 

content.DataType is empty, so assigning "" to it won't change anything

you can assing DataType field directly in `content` struct initialization

Reply:
Moved dataType to content initialization and the very initialization to AddContent call